File: wlmedia_sample/wlmedia/src/main/java/com/ywl5320/wlmedia/util/WlTimeUtil.java
package com.ywl5320.wlmedia.util;

/**
 * author : ywl5320
 * e-mail : ywl5320@163.com
 * desc   : wlmedia
 * date   : 2024/3/23
 */
public class WlTimeUtil {

    static StringBuilder time = new StringBuilder();

    /**
     * format times
     *
     * @param second
     * @return
     */
    public static String secondToTimeFormat(double second) {

        if (second < 0) {
            second = 0;
        }

        if (time.length() > 0) {
            time.delete(0, time.length());
        }

        long hours = (long) (second / (3600));
        long minutes = (long) ((second % (3600)) / (60));
        long seconds = (long) (second % (60));

        if (hours > 0) {
            time.append((hours >= 10) ? hours : "0" + hours);
            time.append(":");
        }
        time.append((minutes >= 10) ? minutes : "0" + minutes);
        time.append(":");
        time.append((seconds >= 10) ? seconds : "0" + seconds);
        return time.toString();
    }

    /**
     * format times
     *
     * @param second
     * @return
     */
    public static String secondToTimeFormat3(double second) {

        if (second < 0) {
            second = 0;
        }

        if (time.length() > 0) {
            time.delete(0, time.length());
        }

        long hours = (long) (second / (3600));
        long minutes = (long) ((second % (3600)) / (60));
        long seconds = (long) (second % (60));

        time.append((hours >= 10) ? hours : "0" + hours);
        time.append(":");
        time.append((minutes >= 10) ? minutes : "0" + minutes);
        time.append(":");
        time.append((seconds >= 10) ? seconds : "0" + seconds);
        return time.toString();
    }
}

File: wlmedia_hm/libwlmedia/src/main/ets/com/ywl5320/wlmedia/util/WlTimeUtil.ts
/**
 * author : ywl5320
 * e-mail : ywl5320@163.com
 * desc   : wlmedia
 * date   : 2024/12/29
 */
export class WlTimeUtil {
  /**
   * Format time from seconds to HH:mm:ss or mm:ss format
   * @param second - Time in seconds
   * @returns Formatted time string
   */
  public static secondToTimeFormat(second: number): string {
    if (second < 0) {
      return "00:00"
    }

    let time: string = ""; // Clear the existing time string

    const hours: number = Math.floor(second / 3600);
    const minutes: number = Math.floor((second % 3600) / 60);
    const seconds: number = Math.floor(second % 60);

    if (hours > 0) {
      time += (hours >= 10 ? hours : "0" + hours) + ":";
    }

    time += (minutes >= 10 ? minutes : "0" + minutes) + ":";
    time += (seconds >= 10 ? seconds : "0" + seconds);

    return time;
  }

  /**
   * Format time from seconds to HH:mm:ss format
   * @param second - Time in seconds
   * @returns Formatted time string
   */
  public static secondToTimeFormat3(second: number): string {
    if (second < 0) {
      return "00:00:00"
    }

    let time: string = "";

    const hours: number = Math.floor(second / 3600);
    const minutes: number = Math.floor((second % 3600) / 60);
    const seconds: number = Math.floor(second % 60);

    time += (hours >= 10 ? hours : "0" + hours) + ":";
    time += (minutes >= 10 ? minutes : "0" + minutes) + ":";
    time += (seconds >= 10 ? seconds : "0" + seconds);

    return time;
  }
}
